Makefile

```diff
@@ -51,19 +51,40 @@ export: VERSION.full
 	rm bats.tar
 	ls -lh charliecloud-$$(cat VERSION.full).tar.gz
 
-BIN=$(PREFIX)/bin
-DOC=$(PREFIX)/share/doc/charliecloud
+INSTALL_PREFIX := $(if $(DESTDIR),$(DESTDIR),$(PREFIX))
+BIN=$(INSTALL_PREFIX)/bin
+LIBEXEC=$(INSTALL_PREFIX)/libexec/charliecloud
+LIBEXEC_RUN=$(PREFIX)/libexec/charliecloud
+DOC=$(INSTALL_PREFIX)/share/doc/charliecloud
```

Comment:
Can you add a comment documenting all these variables, for folks with a shaky understanding of them like myself?

This could be as simple as a link to someone else's write-up.

Reply:
Yep, I'll give that a go. The naming `PREFIX` and `DESTDIR` are modelled close to what they mean for autotools, I'll see if I can find a nice writeup, and otherwise cook up something compact.

Reply:
For reference, Debian's interpretation of FHS is to use `lib` not `libexec` (I think FHS 2.3 doesn't have `/usr/libexec`)

Reply:
@mcv21 Thanks, indeed! Yes, it's only an (optional) part of FHS 3.0 I believe. 
It's already adopted on (at least) Gentoo and RedHat (deducing from a CentOS 7.4 system). 
So I guess there are two options now:

- Use `libexec` here, and change things in the Debian packaging code to use `lib`. 
- Default to `libexec` and add another variable to `make install` which allows to change that to something else (so Debian packaging code could specify `lib`). 

Do you have a preference?

Reply:
Something that the build process could override on the command line would be easiest, but `dh-install` can move things around if necessary.

Reply:
Ok - then I think I'd go with the environment variable. 
The problem with `dh-install` would be that it would not only have to move things around, but then (again) have to patch the script-files to refer to the different libexec-location to find `base.sh`.

Reply:
Good point! Sorry, should have thought of that.

Reply:
...though thinking about this, debian builds work (conventionally) by setting prefix to debian/tmp and then fishing the build results out of there...

Reply:
> ...though thinking about this, debian builds work (conventionally) by setting prefix to debian/tmp and then fishing the build results out of there...

Are you sure they set `prefix`? As I know it from Gentoo and RedHat, they usually use `destdir` for the temporary install location, but specify `prefix` already to the expected future runtime location (i.e. as if things would be installed on the real filesystem). 
This allows to use `prefix` to derive things as we do here ("prepare for run-time location", i.e. put paths in scripts), while `destdir` would then be the build-time sandbox (`debian/tmp`) to which the files are actually copied. 

In any case, I guess I should do my attempt at Debian packaging first before finalizing this PR to be sure - up to now, I never did a Debian package from scratch, but only modified existing ones.

Reply:
Sounds good. I'll start on debian packaging tonight, let's see how far I get and whether I note other changes / subleties from this.

Reply:
I'm almost through with having a working (but not yet finished) debian package - and at least I can confirm they pass `DESTDIR` to `make` in the `install` step, as all other distros do in packaging. 
So at least there are no new surprises on this end.